### ISSUE TITLE
Add tau optional parameter to cumulative_dynamic_auc

### ISSUE THREAD

**Issue by jacobod** [sebp/scikit-survival#191]
**Problem**
For `concordance_index_ipcw` there is an optional parameter `tau` which truncates the censoring estimation function at a given time. This is great for cases where there are durations in the training set with 1 or more events, but the cross validation set has 0 events. It allows the censoring estimation prediction in the cross validation set to truncate before these 0-event times to get a calculation of the adjusted c-index.

However, the same censoring estimation logic is used for `cumulative_dynamic_auc` without a `tau` parameter, so passing in a dataset that would fail the `ipcw` calculation without a `tau` parameter would always result in an error in this function.

**Possible Solutions**

I think there should be some truncation logic applied to the `survival_test` variable similar to `concordance_index_ipcw` to handle this case.
Current logic in that function:
```
    if tau is not None:
        mask = test_time < tau
        survival_test = survival_test[mask]
```

Here are 2 potential solutions:

1. Use the last value in the `times` param to truncate the `survival_test` variable, making the presumption the user does not care about times after `times[-1]`. This will also still throw an error if the user selects a time where there are 0 events in the `survival_test` variable as before.

2. Add an optional parameter `tau` as in `concordance_index_ipcw` and apply the same logic there to the `survival_test` parameter. 

**References**

Current implementation of `concordance_index_icpw`: https://github.com/sebp/scikit-survival/blob/66deb8204e3d8632176d171c8d9dbfae2c06c25e/sksurv/metrics.py#L309

Current implementation of `cumulative_dynamic_auc`: https://github.com/sebp/scikit-survival/blob/66deb8204e3d8632176d171c8d9dbfae2c06c25e/sksurv/metrics.py#L330

**Comment by sebp** [comment]
I'm not sure I fully understood your point. The IPCW are only used for weighting cases, but if your test data has no events for the time points > tau, IPCW has no effect. See https://github.com/sebp/scikit-survival/blob/66deb8204e3d8632176d171c8d9dbfae2c06c25e/sksurv/metrics.py#L471

**Comment by jacobod** [comment]
@sebp thank you for your response! This is correct, and I figured out my issue.

I had a lone event at a duration in the holdout set that was later than the max duration that experienced an event in my training set. Accounting for this allowed the AUC calculation to work as intended.
